luaui/Widgets/gui_unit_market.lua

```diff
@@ -0,0 +1,480 @@
+-- b) write in chat: /sell_unit
+-- c) bind a hotkey before hand: bind alt+c /sell_unit, then just press the hotkey.
+-- d) press button "Sell Unit" at the bottom center part of your screen.
+-- It works like a toggle, so you can toggle selling status by using any of the methods above.
+-- Once you are done you (and your allies) will see that the unit is flashing green and is displaying the sign "$" above itself. The price is exactly the same as the unit metalCost.
+
+-- As a buyer:
+-- 1) hold alt and double-click over an ally unit that your ally is selling. make sure you have resources first. in case you don't have enough metal - nothing will happen.
+
+--------------------------------
+local spGetPlayerInfo       = Spring.GetPlayerInfo
+local spGetSpectatingState  = Spring.GetSpectatingState
+local spGetTeamInfo         = Spring.GetTeamInfo
+local spGetUnitDefID        = Spring.GetUnitDefID
+local spGetUnitTeam			= Spring.GetUnitTeam
+local spGetSelectedUnits    = Spring.GetSelectedUnits
+local spAreTeamsAllied      = Spring.AreTeamsAllied
+local spGetUnitPosition     = Spring.GetUnitPosition
+local spSendLuaUIMsg        = Spring.SendLuaUIMsg
+local spSendLuaRulesMsg     = Spring.SendLuaRulesMsg
+local spValidUnitID         = Spring.ValidUnitID
+local spGetCameraState      = Spring.GetCameraState
+local spGetGameSeconds      = Spring.GetGameSeconds
+local spEcho                = Spring.Echo
+local spLog                 = Spring.Log
+local spIsUnitInView        = Spring.IsUnitInView
+local spGetPlayerList       = Spring.GetPlayerList
+local spGetAIInfo           = Spring.GetAIInfo
+local spGetUnitViewPosition = Spring.GetUnitViewPosition
+local spTraceScreenRay      = Spring.TraceScreenRay
+local spGetUnitsInCylinder  = Spring.GetUnitsInCylinder
+local spGetModKeyState      = Spring.GetModKeyState
+local spGetMouseState       = Spring.GetMouseState
+local spGetUnitRulesParam  	= Spring.GetUnitRulesParam
+local spSetUnitRulesParam   = Spring.SetUnitRulesParam
+local spGetGameRulesParam   = Spring.GetGameRulesParam
+local myTeamID     = Spring.GetMyTeamID()
+local myAllyTeamID = Spring.GetMyAllyTeamID()
+local gaiaTeamID   = Spring.GetGaiaTeamID()
+local isSpectating, fullview = Spring.GetSpectatingState()
+local myPlayerID   = Spring.GetMyPlayerID()
+local logging    = false -- Logging...
+local see_prices = false -- Set to true for local testing to verify unit prices
+local see_sales  = true  -- Set to false to never see console trade messages
+
+local unitMarket = Spring.GetModOptions().unit_market
+local unitsForSale = {} -- Array to store units offered for sale {UnitID => metalCost}
+
+-- button vars
+local sellUnitText = Spring.I18N('ui.unitMarket.sellUnit') or "Sell Unit"
+local buttonPosX = 0.41
+local buttonPosY = 0.066
+local UiButton, UiElement
+local fontfile = "fonts/" .. Spring.GetConfigString("bar_font2", "Exo2-SemiBold.otf")
+local vsx, vsy = Spring.GetViewGeometry()
+local fontfileScale = (0.5 + (vsx * vsy / 6200000))
```

Comment:
Font scale should be set based on UI scale, you can use `gui_gridmenu` for reference on this